doc/develop/test/twister/twister_blackbox.rst

```diff
@@ -0,0 +1,119 @@
+.. _twister_blackbox:
+
+Twister blackbox tests
+######################
+
+This guide aims to explain the structure of a test file so the reader will be able
+to understand existing files and create their own. All developers should fix any tests
+they break and create new ones when introducing new features, so this knowledge is
+important for any Twister developer.
+
+Basics
+******
+
+Twister blackbox tests are written in python, with auxiliary test data following
```

Comment:
I think it is essential to mention `pytest` and refer to its doc pages wherever it is helpful.
And it should be clear how it relates to the pytest Twister plugin https://docs.zephyrproject.org/latest/develop/test/pytest.html